qhbmlib/inference/qnn.py

```diff
@@ -91,6 +103,39 @@ def circuit(self):
   def differentiator(self):
     return self._differentiator
 
+  def _sampled_expectation(self, initial_states: tf.Tensor,
+                           observable: hamiltonian.Hamiltonian):
+    """Returns the expectation values of the observables against the QNN.
+
+    Args:
+      initial_states: Shape [batch_size, num_qubits] of dtype `tf.int8`.
+        Each entry is an initial state for the set of qubits.  For each state,
+        `qnn` is applied and the pure state expectation value is calculated.
+      observables: Hermitian operator to measure.  Will be tiled to measure
+        `<op>_((qnn)|initial_states[i]>)` for each i.
```

Comment:
perhaps add some words for those unfamiliar with the notation e.g. "the expectation value of the observable in the state `qnn|initial_states[i]>` batched over `i`"

Reply:
Done